src/components/SignInButton/SignInButton.tsx

```diff
@@ -0,0 +1,56 @@
+import { useEthereum } from '../../contexts/EthereumContext'
+import { useNetwork } from '@/contexts/NetworkContext'
+import { useProfile } from '../../contexts/ProfileContext'
+
+/**
+ * Provides a button for signing in with a wallet or Universal
+ * Profile and proofing ownership over an address. It leverages
+ * the useEthereum hook from the EthereumContext
+ * for managing blockchain connections.
+ *
+ * @returns {JSX.Element} - A JSX structure of the button
+ */
+function SignInButton() {
+  const { account, verified, signInWithEthereum } = useEthereum()
+  const { network } = useNetwork()
+  const { profile } = useProfile()
+
+  let buttonText
+  let buttonAction
+  let buttonClass
+  let disabled
+
+  if (!account || !network) {
+    // User is not connected or on the wrong network
+    buttonText = 'Sign in with UP'
+    buttonAction = null // No action
```

Comment:
What is the point of having a button which has no action? This is confusing.
It says "Sign in with UP" but when you will click nothing will happen.

(I know the PR is draft so maybe this is not done yet but im flagging still)